zerver/tests/test_integrations_dev_panel.py

```diff
@@ -0,0 +1,75 @@
+import ujson
+from mock import MagicMock, patch
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.models import get_user, get_realm
+
+class TestIntegrationsDevPanel(ZulipTestCase):
+
+    zulip_realm = get_realm("zulip")
+
+    def test_check_send_webhook_fixture_message_for_error(self) -> None:
+        bot = get_user('webhook-bot@zulip.com', self.zulip_realm)
+        url = "/api/v1/external/airbrake?api_key={key}".format(key=bot.api_key)
+        target_url = "/devtools/integrations/check_send_webhook_fixture_message"
+        body = "{}"  # This empty body should generate a KeyError on the webhook code side.
+
+        data = {
+            "url": url,
+            "body": body
+        }
+
+        response = self.client_post(target_url, data)
+
+        self.assertEqual(response.status_code, 500)  # Since the response would be forwarded.
+        expected_response = {"result": "error", "msg": "Internal server error"}
+        self.assertEqual(ujson.loads(response.content), expected_response)
+
+    def test_check_send_webhook_fixture_message_for_success(self) -> None:
+        bot = get_user('webhook-bot@zulip.com', self.zulip_realm)
+        url = "/api/v1/external/airbrake?api_key={key}".format(key=bot.api_key)
+        target_url = "/devtools/integrations/check_send_webhook_fixture_message"
+        with open("zerver/webhooks/airbrake/fixtures/error_message.json", "r") as f:
+            body = f.read()  # This empty body should generate a KeyError on the webhook code side.
+
+        data = {
+            "url": url,
+            "body": body
+        }
+
+        response = self.client_post(target_url, data)
+        self.assertEqual(response.status_code, 200)
```

Comment:
I would also check that the last message matches what it should be (at least to the target stream/topic)

Reply:
I've upgraded this test. :+1: